baseapp/abci.go

```diff
@@ -426,10 +437,13 @@ func (app *BaseApp) Commit() abci.ResponseCommit {
 		RetainHeight: retainHeight,
 	}
 
-	// call the hooks with the Commit message
-	for _, streamingListener := range app.abciListeners {
-		if err := streamingListener.ListenCommit(app.deliverState.ctx, res); err != nil {
-			panic(fmt.Errorf("Commit listening hook failed, height: %d, err: %w", header.Height, err))
+	// call the streaming service hook with the EndBlock messages
+	for _, abciListener := range app.streamingManager.AbciListeners {
+		ctx := app.deliverState.ctx
+		blockHeight := ctx.BlockHeight()
+		changeSet := app.cms.PopStateCache()
+		if err := abciListener.ListenCommit(ctx, res, changeSet); err != nil {
+			app.logger.Error("Commit listening hook failed", "height", blockHeight, "err", err)
 		}
```

Comment:
You're calling app.cms.PopStateCache for every AbciListener, but PopStateCache is a destructive operation, which invalidates the returned value of the app.cms. This means that only the first streamingManager.AbciListener will get a valid changeSet.

Where is this method tested? How did this bug survive to this PR?

Reply:
Looks like we all missed it during ADR review. I added unit testing and fixed the call so all listeners receive a valid changeSet.

Reply:
Going to take a second look and improve it a bit

Reply:
done.